src/build_vehicle_gui.cpp

```diff
@@ -854,28 +858,41 @@ static int DrawAircraftPurchaseInfo(int left, int right, int y, EngineID engine_
 	return y;
 }
 
+
 /**
- * Display additional text from NewGRF in the purchase information window
- * @param left   Left border of text bounding box
- * @param right  Right border of text bounding box
- * @param y      Top border of text bounding box
- * @param engine Engine to query the additional purchase information for
- * @return       Bottom border of text bounding box
+ *
  */
-static uint ShowAdditionalText(int left, int right, int y, EngineID engine)
+static std::tuple<bool, std::string> GetNewGRFAdditionalText(EngineID engine)
 {
 	uint16 callback = GetVehicleCallback(CBID_VEHICLE_ADDITIONAL_TEXT, 0, 0, engine, nullptr);
-	if (callback == CALLBACK_FAILED || callback == 0x400) return y;
+	if (callback == CALLBACK_FAILED || callback == 0x400) return std::make_tuple(false, "");
 	const GRFFile *grffile = Engine::Get(engine)->GetGRF();
 	assert(grffile != nullptr);
 	if (callback > 0x400) {
 		ErrorUnknownCallbackResult(grffile->grfid, CBID_VEHICLE_ADDITIONAL_TEXT, callback);
-		return y;
+		return std::make_tuple(false, "");
 	}
 
 	StartTextRefStackUsage(grffile, 6);
-	uint result = DrawStringMultiLine(left, right, y, INT32_MAX, GetGRFStringID(grffile->grfid, 0xD000 + callback), TC_BLACK);
+	std::string result = GetString(GetGRFStringID(grffile->grfid, 0xD000 + callback));
 	StopTextRefStackUsage();
+	return std::make_tuple(true, result);
+}
+
+/**
+ * Display additional text from NewGRF in the purchase information window
+ * @param left   Left border of text bounding box
+ * @param right  Right border of text bounding box
+ * @param y      Top border of text bounding box
+ * @param engine Engine to query the additional purchase information for
+ * @return       Bottom border of text bounding box
+ */
+static uint ShowAdditionalText(int left, int right, int y, EngineID engine)
+{
+	std::tuple<bool, std::string> packed = GetNewGRFAdditionalText(engine);
```

Comment:
In general, with tuples and the likes you more likely want to use a structured binding declaration like:
`auto& [succeeded, text] = GetNewGRFAdditionalText(engine);` as figuring out the indices later on when unpacking is not that nice.